File: Entities/Secret.cs
using System;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using SecretShare.Entities.Interfaces;
using SecretShare.Entities;
using System.ComponentModel.DataAnnotations.Schema;

namespace SecretShare.Entities
{

    /// <summary>
    /// A secret is a submitted secret for bucket.
    /// It contains the encrypted data to be decrypted by the bucket owner. 
    /// </summary>
    /// <value></value>
    public record Secret : IDateTracked
    {
        /// <summary>
        /// The id of this secret.
        /// </summary>
        public Guid SecretId { get; init; }

        /// <summary>
        /// Related bucket
        /// </summary>
        [ForeignKey("BucketId")]
        [Required]
        public Bucket Bucket { get; set; }

        /// <summary>
        /// A title/description of the secret
        /// </summary>
        [Column(TypeName = "varchar(255)")]
        public String Title { get; init; }

        /// <summary>
        /// The encrypted value
        /// </summary>
        public String Value { get; init; }

        /// <summary>
        /// Flag for viewed status
        /// </summary>
        public bool Viewed { get; init; }

        /// <summary>
        /// Date when viewed
        /// </summary>
        public DateTime? ViewedAt { get; init; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
    }
}

File: frontend/src/lib/models/Secret.ts
import { SecretDto } from '../dtos/Response/secretDto'

export class Secret implements SecretDto {
  secretDto: any
  private _secretId: string = ''
  public get secretId(): string {
    return this._secretId
  }
  public set secretId(value: string) {
    this._secretId = value
  }
  private _title: string = ''
  public get title(): string {
    return this._title
  }
  public set title(value: string) {
    this._title = value
  }
  private _value: string = ''
  public get value(): string {
    return this._value
  }
  public set value(value: string) {
    this._value = value
  }
  private _viewed: boolean = false
  public get viewed(): boolean {
    return this._viewed
  }
  public set viewed(value: boolean) {
    this._viewed = value
  }
  private _viewedAt?: Date
  public get viewedAt(): Date | undefined {
    return this._viewedAt
  }
  public set viewedAt(value: Date | undefined) {
    this._viewedAt = value
  }
  private _createdAt: Date = new Date()
  public get createdAt(): Date {
    return this._createdAt
  }
  public set createdAt(value: Date) {
    this._createdAt = value
  }
  private _updatedAt: Date = new Date()
  public get updatedAt(): Date {
    return this._updatedAt
  }
  public set updatedAt(value: Date) {
    this._updatedAt = value
  }
}
